include/s3select.h

```diff
@@ -2111,25 +2115,36 @@ class csv_object : public base_s3object
 public:
   int run_s3select_on_object(std::string& result, const char* csv_stream, size_t stream_length, bool skip_first_line, bool skip_last_line, bool do_aggregate)
   {
-
-
-    m_stream = (char*)csv_stream;
-    m_end_stream = (char*)csv_stream + stream_length;
+    m_stream << (char*)csv_stream;
+    m_end_stream << ((char*)csv_stream + stream_length);
     m_is_to_aggregate = do_aggregate;
     m_skip_last_line = skip_last_line;
 
+    // Count columns
+    for(auto i = 0; i < (int)strlen(csv_stream); i++)
```

Comment:
not sure it is correct.
`csv_stream` may point to a broken line, thus the `m_column_count` may not be correct